### PR TITLE
Item 6199: Domain designer test updates for changes with assay designer

### PR DESCRIPTION
**Pull request by cnathe**

### CODE REVIEW

File: src/org/labkey/test/pages/ReactAssayDesignerPage.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    }
+
+    public void clickFinish()
+    {
+        scrollIntoView(elementCache().finishButton);
+        shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().finishButton));
+        String currentURL = getDriver().getCurrentUrl();
+        elementCache().finishButton.click();
+        afterSaveOrFinishClick(currentURL);
+    }
```
Comment (labkey-tchad) [inline src/org/labkey/test/pages/ReactAssayDesignerPage.java]: I'm more convinced that these wizard workflow methods belong in a separate class. The pattern that is arising from this design is going to be painful to maintain; specifically, having to know the name of the last domain panel before clicking finish.

File: src/org/labkey/test/components/domain/DomainFieldRow.java [hunk 2]
```diff
@@ -86,6 +86,7 @@ public WebElement typeInput()
 
     public DomainFieldRow setType(String columnType)
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/components/domain/DomainFieldRow.java]: Done.

File: src/org/labkey/test/components/domain/DomainFormPanel.java [hunk 3]
```diff
@@ -79,9 +81,20 @@ public DomainFieldRow addField(String name)
...
+
+        DomainFieldRow newFieldRow = elementCache().findFieldRows().get(0);
+        newFieldRow.setName(name);
+        return newFieldRow;
+    }
+
     public DomainFormPanel removeField(String name)
     {
-        getField(name).clickRemoveField().dismiss("Yes");
+        getField(name).expand().clickRemoveField().dismiss("Yes");
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/components/domain/DomainFormPanel.java]: Good catch. That is done in clickRemoveField().

File: src/org/labkey/test/components/domain/DomainFormPanel.java [hunk 4]
```diff
@@ -79,9 +81,20 @@ public DomainFieldRow addField(String name)
...
+    public DomainFieldRow startNewDesign(String name)
+    {
+        getWrapper().scrollIntoView(elementCache().startNewDesignLink, true);
+        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().startNewDesignLink)); // give modal dialogs time to disappear
+        elementCache().startNewDesignLink.click();
+
+        DomainFieldRow newFieldRow = elementCache().findFieldRows().get(0);
+        newFieldRow.setName(name);
+        return newFieldRow;
+    }
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/components/domain/DomainFormPanel.java]: Right now the only domain that has this "default to the file drop state" is the assay results domain. However, the plan is to use this in more places (for example lists, datasets) once we start migrating those UI components. So I believe in that case that DomainFormPanel is the right place for this.

File: src/org/labkey/test/components/domain/DomainFormPanel.java [hunk 5]
```diff
@@ -171,21 +190,29 @@ private DomainFieldRow findFieldRow(String name)
...
+                .refindWhenNeeded(this).withTimeout(WAIT_FOR_JAVASCRIPT);
+        WebElement expandIcon = Locator.tagWithClass("svg", "fa-plus-square")
+                .refindWhenNeeded(this).withTimeout(WAIT_FOR_JAVASCRIPT);
     }
 
     public static class DomainFormPanelFinder extends WebDriverComponentFinder<DomainFormPanel, DomainFormPanelFinder>
     {
-        private final Locator.XPathLocator _baseLocator = Locator.tagWithClass("div", "domain-form-panel");
+        final Locator.XPathLocator _baseLocator = Locator.tagWithClass("div", "domain-form-panel");
+        final Locator.XPathLocator _activeLocator = Locator.tagWithClass("div", "domain-form-panel").withClass("panel-active");
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/components/domain/DomainFormPanel.java]: Agreed. The single domain editing pages don't have this notion of "active" panel, but once we start migrating the lists/datasets/issues domains, I think we will have that case there as well. 

I see what you are saying for moving the panel expand/collapse stuff out to an abstract class. I'll take a look, but in the interest of getting this story closed out for 19.3, I might defer and put a TODO in the test class.

File: src/org/labkey/test/components/domain/DomainFormPanel.java [hunk 6]
```diff
@@ -171,21 +190,29 @@ private DomainFieldRow findFieldRow(String name)
...
         private String _title = null;
+        private boolean _active;
 
         public DomainFormPanelFinder(WebDriver driver)
         {
             super(driver);
         }
 
-        public DomainFormPanelFinder withTitle(String title)
+        public DomainFormPanelFinder withTitle(String title, boolean active)
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/components/domain/DomainFormPanel.java]: Good call. Changed.

File: src/org/labkey/test/pages/ReactAssayDesignerPage.java [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+    {
+        elementCache().backgroundUploadCheckbox.set(checked);
+        return this;
+    }
+
+    public ReactAssayDesignerPage enableQCStates(boolean checked)
+    {
+        elementCache().qcEnabledCheckbox.set(checked);
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/pages/ReactAssayDesignerPage.java]: Done.

File: src/org/labkey/test/pages/ReactAssayDesignerPage.java [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+import org.openqa.selenium.support.ui.Select;
+
+import java.io.File;
+
+import static org.junit.Assert.assertTrue;
+import static org.labkey.test.components.html.Checkbox.Checkbox;
+import static org.labkey.test.components.html.Input.Input;
+import static org.labkey.test.components.html.SelectWrapper.Select;
+
+public class ReactAssayDesignerPage extends DomainDesignerPage
```
[the remark this answers is not in the document]
Comment (cnathe) [reply src/org/labkey/test/pages/ReactAssayDesignerPage.java]: Done.

**Pull request merged**